var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -110,8 +110,27 @@ def install(self, spec, prefix):
                         '-rhsone')
             make("install")
 
+    @property
+    def headers(self):
+        """Export the main hypre header, HYPRE.h; all other headers can be found
+        in the same directory.
+        Sample usage: spec['hypre'].headers.cpp_flags
+        """
+        hdrs = HeaderList(find(self.prefix.include, 'HYPRE.h',
+                               recursive=False))
+        return hdrs or None
+
     @property
     def libs(self):
-        is_shared = self.spec.satisfies('+shared')
-        return find_libraries('libHYPRE', root=self.prefix,
-                              shared=is_shared, recursive=True)
+        """Export the hypre library.
+        Sample usage: spec['hypre'].libs.ld_flags
+        """
+        search_paths = [[self.prefix.lib, False], [self.prefix.lib64, False],
+                        [self.prefix, True]]
+        is_shared = '+shared' in self.spec
```

Comment:
I also use this a lot, and I think it's fine. But, if we have to be subtle,  this means that at least one spec in the DAG as the `+shared` variant.

Reply:
Actually, in the current implementation, this is not true. I tested this with some debug prints: in `Spec.__contains__` the string `'+shared'` is turned (by the `_autospec` call) into a `Spec`: `'hypre+shared'` (printed as a string). So, even though the dependencies are traversed, they will not satisfy the spec `'hypre+shared'`.

Anyway, for me, this confirms that `'+shared' in self.spec` has the right behavior.